diff -ur a/src/libponyc/expr/array.c b/src/libponyc/expr/array.c
--- a/src/libponyc/expr/array.c	2017-09-14 03:28:30.000000000 +0200
+++ b/src/libponyc/expr/array.c	2017-09-14 16:55:29.128717701 +0200
@@ -149,7 +149,7 @@
 
       // Note this as a possible element type and move on.
       *list = astlist_push(*list, elem_type);
-    }
+    } // fall through
 
     case TK_ARROW:
       find_possible_element_types(opt, ast_childidx(ast, 1), list);
@@ -190,7 +190,7 @@
         *list = astlist_push(*list, ast_child(typeargs));
         return;
       }
-    }
+    } // fall through
 
     case TK_ARROW:
       find_possible_iterator_element_types(opt, ast_childidx(ast, 1), list);
