diff -ur a/src/libponyc/codegen/genopt.cc b/src/libponyc/codegen/genopt.cc
--- a/src/libponyc/codegen/genopt.cc	2017-06-25 12:58:03.426580762 +0200
+++ b/src/libponyc/codegen/genopt.cc	2017-06-25 12:56:23.764481259 +0200
@@ -450,7 +450,10 @@
   PassManagerBase& pm)
 {
   if(pmb.OptLevel >= 2)
+  {
+    pm.add(new DominatorTreeWrapperPass());
     pm.add(new HeapToStack());
+  }
 }
 
 class DispatchPonyCtx : public FunctionPass
