diff -ur a/Makefile b/Makefile
--- a/Makefile	2017-06-25 12:51:06.489791633 +0200
+++ b/Makefile	2017-06-25 13:00:28.930886129 +0200
@@ -437,11 +437,11 @@
 libponyrt.benchmarks.links = libgbenchmark libponyrt
 
 ifeq ($(OSTYPE),linux)
-  ponyc.links += libpthread libdl
-  libponyc.tests.links += libpthread libdl
-  libponyrt.tests.links += libpthread libdl
-  libponyc.benchmarks.links += libpthread libdl
-  libponyrt.benchmarks.links += libpthread libdl
+  ponyc.links += libpthread libdl libatomic
+  libponyc.tests.links += libpthread libdl libatomic
+  libponyrt.tests.links += libpthread libdl libatomic
+  libponyc.benchmarks.links += libpthread libdl libatomic
+  libponyrt.benchmarks.links += libpthread libdl libatomic
 endif
 
 ifeq ($(OSTYPE),freebsd)
diff -ur a/src/libponyc/codegen/genexe.c b/src/libponyc/codegen/genexe.c
--- a/src/libponyc/codegen/genexe.c	2017-05-06 12:00:54.000000000 +0200
+++ b/src/libponyc/codegen/genexe.c	2017-06-25 13:01:43.925951031 +0200
@@ -277,7 +277,7 @@
 #ifdef PONY_USE_LTO
     "-flto -fuse-linker-plugin "
 #endif
-    "%s %s %s %s -lpthread %s -lm %s",
+    "%s %s %s %s -lpthread -latomic %s -lm %s",
     linker, file_exe, arch, mcx16_arg, fuseld, file_o, lib_args, ponyrt, ldl,
     export
     );
